src/main.py

```diff
@@ -1,125 +1,246 @@
 # Leah
+# main.py
 # Written by aquova et al., 2022
-# https://github.com/aquova/leah
+# https://github.com/StardewValleyDiscord/leah
 
+import random
+import strings
 import discord
-from config import client, DISCORD_KEY,\
-    ART_CHANS, MOD_CHANS, VERIFY_CHAN, GALLERY_CHAN, SHOWCASE_CHAN,\
-    SHOWCASE_ROLES
+from discord.ext import commands
+from config import COMMAND_PREFIX, EXTENSIONS, DISCORD_KEY, DISCORD_INTENTS, \
+    ART_CHANS, MOD_CHANS, VERIFY_CHAN, GALLERY_CHAN, SHOWCASE_CHAN, VERIFY_ROLES, SHOWCASE_ROLES, ADMIN_ROLE
+from importlib import reload
 from typing import Optional
+from utils import check_roles, format_roles_error
 
-class Leah:
+
+# Bot definition
+class Leah(commands.Bot):
     def __init__(self):
-        self.posted = set()
+        super().__init__(
+            command_prefix=COMMAND_PREFIX,
+            intents=DISCORD_INTENTS,
+            description=strings.get("client_description"),
+            allowed_mentions=discord.AllowedMentions.none())
+
+    async def sync_guild(self, guild: discord.Guild):
+        self.tree.copy_global_to(guild=guild)
+        await self.tree.sync(guild=guild)
+
+    async def setup_hook(self) -> None:
```

Comment:
Do we need to call this somewhere?

Reply:
Nope! It's a method of `discord.ext.commands.Bot` that's called internally after login:

https://discordpy.readthedocs.io/en/stable/api.html#discord.Client.setup_hook

Reply:
Some comments on the bot's methods wouldn't be amiss though so I'll stick those on too.